### PR TITLE
Optimize `get_td_waveform`

### PR DESCRIPTION
**Pull request by vaishakp**
Interpolate only once, after evaluation instead of interpolating all modes before evaluation.

To be merged after #32, #35 and #33.

### CODE REVIEW

File: nrcatalogtools/waveform.py [hunk 1]
```diff
@@ -274,9 +274,9 @@ def get_td_waveform(
             inclination=inclination, coa_phase=coa_phase, f_ref=f_ref, t_ref=t_ref
         )
 
-        h = self.interpolate(new_time).evaluate(
+        h = sel.evaluate(
             [angles["theta"], angles["psi"], angles["alpha"]]
-        ) * utils.amp_to_physical(total_mass, distance)
+        ).interpolate(new_time) * utils.amp_to_physical(total_mass, distance)
```
[the remark this answers is not in the document]
Comment (prayush) [reply nrcatalogtools/waveform.py:279]: Never mind, I realized that in the first case, we interpolate all modes, instead of just the polarizations.

**Review by prayush — APPROVED** [review]

File: nrcatalogtools/waveform.py [hunk 2]
```diff
@@ -274,9 +274,9 @@ def get_td_waveform(
             inclination=inclination, coa_phase=coa_phase, f_ref=f_ref, t_ref=t_ref
         )
 
-        h = self.interpolate(new_time).evaluate(
+        h = sel.evaluate(
```
[the remark this answers is not in the document]
Comment (prayush) [reply nrcatalogtools/waveform.py]: BTW, how did this typo not lead to any test failure.

[the remark this answers is not in the document]
Comment (vaishakp) [reply nrcatalogtools/waveform.py]: Currently we only compare modes. With #33, this would failed the test pipeline.

**Comment by vaishakp** [comment]
> @vaishakp Is there a reason this should wait for #32, #33 and #35 (this seems ready to go now)?

Actually no. This can be merged.

**Pull request merged**